### PR TITLE
Chore/#tc 13 add dockerfile for admin page

### PR DESCRIPTION
**Pull request by DryDish**
### This pull request closes:

✔️ Closes: N/A
🏁 Jira Issue: [TC-13](https://treecreate.atlassian.net/browse/TC-19)

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [ ] 🛒 webstore
- [x] 🎓 admin-page

### Types of changes

- [x] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [ ] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)

### Documentation Updates

- [x] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [ ] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [ ] Tested on Safari
- [ ] Tested on Chrome
- [ ] Tested on Mozilla Firefox

### What changes have been done within this issue?

* Changed webstore's Dockerfile to use new system.
* Changed package.json's ` postinstall` tag
* ~Updated both Dockerfiles to retrieve angular-cli as part of the installation process~ no longer needed
* Moved Angular build dependencies out of devDependecies to regular dependencies
### How should this be manually tested?

Run the project and Dockerfiles

docker build -t webstore:latest -f .deploy/webstore/Dockerfile .
docker run --name webstore -p 4200:80 -e API_URL:https://api.testing.treecreate.dk  webstore:latest

-> navigate to http://localhost:4200/dk/home

### CODE REVIEW

File: .deploy/admin-page/Dockerfile [hunk 1]
```diff
@@ -34,10 +34,6 @@ COPY nx.json ./nx.json
 COPY tsconfig.base.json ./tsconfig.base.json
 COPY workspace.json ./workspace.json
 
-# Install NX since it is not installed in our repository
-# If we skip this step the prod build fails, as it is not a prod dependency
-RUN npm install nx
-
```
Comment (DryDish) [inline .deploy/admin-page/Dockerfile:40]: no longer needed

File: .deploy/webstore/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,24 @@
-FROM node:lts AS builder
+FROM node:lts-alpine3.14 AS builder
```
Comment (DryDish) [inline .deploy/webstore/Dockerfile:4]: smol and zoom

File: package.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
     "workspace-generator": "nx workspace-generator",
     "dep-graph": "nx dep-graph",
     "help": "nx help",
-    "postinstall": "ngcc --properties es2015 browser module main"
+    "postinstall": "ngcc"
```
Comment (DryDish) [inline package.json:54]: Fixed postinstall script

File: .deploy/admin-page/Dockerfile [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
 # Lightweight environment with node pre-installed
 FROM node:lts-alpine3.14 AS builder
 
 # Directory to work in the container
 WORKDIR /app
 
-# Copy package.json to the root of container
-COPY package.json .
-
-# Set required ENVs
```
Comment (DryDish) [inline .deploy/admin-page/Dockerfile:15]: moved under ENV for efficiency

File: package.json [hunk 5]
```diff
@@ -66,37 +66,39 @@
...
     "rxjs": "^7.4.0",
     "tslib": "^2.0.0",
-    "zone.js": "0.11.4"
+    "zone.js": "0.11.4",
+    "@angular/compiler-cli": "13.0.2",
+    "@nrwl/cli": "13.2.1",
+    "@nrwl/nx-cloud": "12.5.4",
+    "@nrwl/workspace": "13.2.1",
+    "@angular-devkit/build-angular": "13.0.3",
+    "typescript": "4.4.4"
```
Comment (DryDish) [inline package.json:80]: These dependencies had to be moved here as they are required to build the prod version of webstore and admin-page

**Comment by DryDish** [comment]
ready for review!

**Review by Kwandes — CHANGES_REQUESTED** [review]
I'm doing extensive testing of the branch making sure everything works correctly
But in terms of code alone, it looks good, and the images build
One note though: You may have gitignored the package-lock.json, you have to manually remove it since it is still tracked (once a file is added to git it gets tracked and has to be manually deleted)
Oh, and update your branch

**Comment by Kwandes** [comment]
When deployed to staging the webstore app fails to load properly:
image
Admin works fine
When running locally via nx or running the docker image locally it works fine
Haven't tested deployed prod for obvious reasons

**Review by Kwandes — APPROVED** [review]
🚀

**Pull request merged**